### PR TITLE
add docs for merchant search

### PR DESCRIPTION
**Pull request by bpal938**
## Description
Add docs for search merchants endpoint

image

## Ticket
https://www.notion.so/centrapay/Store-Finder-Create-search-merchants-endpoint-e3f5f2dabbec4ae185c7b4ae0ceda55e

### CODE REVIEW

**Review by tom-blake — COMMENTED** [review]
@bpal938 We should document the route permission as part of this too

File: legacy/src/api/merchants/merchants.md [hunk 1]
```diff
@@ -271,6 +271,52 @@ Returns a [paginated][] list of Merchants attached to an Account.
...
+  query_param 'name', 'Store'
+{% endreqspec %}
+
+{% h4 Request Query Parameters %}
+
+|       Query Parameter   |           Type            |                                 Description                                                         |
+| :---------------------- | :------------------------ | :--------------------------------------------------------------------------------------------------- |
+| {% break _ origin %}    | {% dt String %} {% opt %} | The point to be searched around, formatted as a latitude,longitude                                   |
+| {% break _ distance %}  | {% dt Number %} {% opt %} | The distance from the origin to be searched around in km                                             |
+| {% break _ assetType %} | {% dt String %} {% opt %} | The assetType to filter merchants by. This can be supplied multiple times to further filter results  |
```
Comment (Neeraj-Ghodla) [inline legacy/src/api/merchants/merchants.md:290]: You should add a link on the `assetType` text which links it to the assetTypes page.

**Comment by bpal938** [comment]
> @bpal938 We should document the route permission as part of this too

@tom-blake We don't currently have any specific permissions for this endpoint, just that it needs auth. Were there specific permissions you are expecting the endpoint to require?

File: legacy/src/api/merchants/merchants.md [hunk 2]
```diff
@@ -271,6 +271,52 @@ Returns a [paginated][] list of Merchants attached to an Account.
...
+
+{% h4 Example response payload %}
+
+{% json %}
+{
+	items: [
+		{
+			"id": "M001",
+			"name": "Store 1",
+			"acceptedAssetTypes": [
```
Comment (tom-blake) [inline legacy/src/api/merchants/merchants.md:301]: @bpal938 This response payload uses a new model that we are not documented anywhere

**Comment by tom-blake** [comment]
> > @bpal938 We should document the route permission as part of this too
> 
> @tom-blake We don't currently have any specific permissions for this endpoint, just that it needs auth. Were there specific permissions you are expecting the endpoint to require?

@bpal938 Yes - we should have a permission for this route. Anyone can sign up to Centrapay and get a JWT

**Review by Neeraj-Ghodla — APPROVED** [review]

File: legacy/src/api/merchants/merchants.md [hunk 3]
```diff
@@ -271,6 +271,59 @@ Returns a [paginated][] list of Merchants attached to an Account.
...
+Returns a [paginated][] list of Merchants with [AssetType][] information
+
+{% reqspec %}
+  GET '/api/merchants/search'
+  auth 'api-key'
+  query_param 'origin', '123.42,32.22'
+  query_param 'distance', '100'
+  query_param 'name', 'Store'
+  query_param 'pageKey' , '2'
+  query_param 'paginationLimit', '1'
```
Comment (Neeraj-Ghodla) [inline legacy/src/api/merchants/merchants.md:285]: Can you change pageinationLimit to something realistic? like 20 or something else

**Pull request merged**